db/agency.go

```diff
@@ -0,0 +1,115 @@
+}
+
+//A Struct containing a snapshot of a agency in a month.
+type AgencyMonthlyInfo struct {
+	AgencyID string     `json:"id" bson:"_id,omitempty"`
+	Storage  []Metadata `json:"storage" bson:"storage,omitempty"`
+	Month    int        `json:"month" bson:"month,omitempty"`
+	Year     int        `json:"year" bson:"year,omitempty"`
+	Summary  Summary    `json:"sumarry" bson:"summary,omitempty"`
+	Employee []Employee `json:"employee" bson:"employee,omitempty"`
+	Metadata Metadata   `json:"metadata" bson:"metadata,omitempty"`
+}
+
+//A Struct containing metadatas about crawler commit
+type Metadata struct {
+	Timestamp      time.Time `json:"metadata" bson:"metadata,omitempty"`     // Time the crawler sent it
+	CrawlerId      string    `json:"crawl_id" bson:"crawl_id,omitempty"`     // The directory of the collector's crawler
+	CrawlerVersion string    `json:"crawl_vers" bson:"crawl_vers,omitempty"` // Last Commit of the repository
+}
+
+// A Struct containing URL to download a file and a hash to track if in the future will be changes in the file.
+type FileBackup struct {
+	URL  string `json:"url" bson:"url,omitempty"`
+	Hash string `json:"hash" bson:"hash,omitempty"`
+}
+
+// A Struct containing summarized  information about a agency/month stats
+type Summary struct {
+	Count  int         `json:"count" bson:"count,omitempty"`   // Number of employees
+	Wage   DataSummary `json:"wage" bson:"wage,omitempty"`     //  Statistics (Max, Min, Median, Total)
+	Perks  DataSummary `json:"perks" bson:"perks,omitempty"`   //  Statistics (Max, Min, Median, Total)
+	Others DataSummary `json:"others" bson:"others,omitempty"` //  Statistics (Max, Min, Median, Total)
+}
+
+// Data Summary with statistics.
+type DataSummary struct {
+	Max   float64 `json:"max" bson:"max,omitempty"`
+	Min   float64 `json:"min" bson:"min,omitempty"`
+	Mean  float64 `json:"mean" bson:"mean,omitempty"`
+	Total float64 `json:"total" bson:"total,omitempty"`
+}
+
+// A Struct that reflets a employee snapshot, containing all relative data about a employee
+type Employee struct {
+	Reg       string        `json:"reg" bson:"reg,omitempty"` // Register number
+	Name      string        `json:"name" bson:"name,omitempty"`
+	Role      string        `json:"role" bson:"role,omitempty"`
+	Type      string        `json:"type" bson:"type,omitempty"`           // servidor, membro, pensionista or indefinido
+	Workplace string        `json:"workplace" bson:"workplace,omitempty"` // 'Lotacao' Like '10° Zona eleitoral'
+	Active    bool          `json:"active" bson:"active,omitempty"`       // 'Active' Or 'Inactive'
+	Income    IncomeDetails `json:"income" bson:"income,omitempty"`
+	Discounts Discount      `json:"discounts" bson:"discounts,omitempty"`
+}
+
+// Struct that details an employee's income.
+type IncomeDetails struct {
+	Total float64  `json:"total" bson:"total,omitempty"`
+	Wage  *float64 `json:"wage" bson:"wage,omitempty"`
+	Perks Perks    `json:"perks" bson:"perks,omitempty"`
```

Comment:
O que acontece se não houver nenhuma Perk? Ou seja, não deveria ser *Perk?

Reply:
Perks e Funds tem a particularidade de que se o campo total for nil, já saberemos que o perk também não terá informação., por isso não vejo necessidade.